### PR TITLE
Update API and add integration testing.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: lib/src/_leak_checker.dart
Comment (polina-c) [inline lib/src/_leak_checker.dart:50]: remove before merging

File: test/flutter/integration/README.md
Comment (CoderDake) [inline test/flutter/integration/README.md:2]:
```suggestion
The tests can serve as examples for regression leak-testing for Flutter widgets.
```

File: test/test_infra/helpers/gc.dart
Comment (CoderDake) [inline test/test_infra/helpers/gc.dart:12]: If we are running these stress tests on our github bots, we may need to double check if we will be taxing any shared quotas.
Do you know if this is something that will take a lot of resources on the bots, or use a lot of time units?

File: lib/src/leak_tracker_model.dart
Comment (CoderDake) [inline lib/src/leak_tracker_model.dart:20]:
```suggestion
  /// leak checking is invoked, will notify only [leakListener].
```

File: test/dart/platform_test.dart
Comment (CoderDake) [inline test/dart/platform_test.dart:28]:
```suggestion
/// Tests for non-guaranteed assumptions about Dart garbage collector,
```

**Review by CoderDake — APPROVED** [review]
LGTM, but I do wonder if the stress tests put any undue stress on the github bots, or our shared resources.
Do you know if that will be an issue?

File: test/test_infra/helpers/gc.dart
Comment (polina-c) [reply test/test_infra/helpers/gc.dart:12]: If everything is as expected, this should not take a lot of resources: the cycle will stop at first garbage collection, that should happen much earlier than any significant impact for bots.

**Comment by polina-c** [comment]
> LGTM, but I do wonder if the stress tests put any undue stress on the github bots, or our shared resources. Do you know if that will be an issue?

If everything is as expected, this should not take noticeable resources: the cycle will stop at first garbage collection, that should happen much earlier than any significant impact for bots.

**Pull request merged**
